### PR TITLE
Redis API Upgrade - 2020-12-01 to 2021-06-01

### PR DESCRIPTION
**Pull request by xuzhang3**
Update Redis API from 2020-12-01 to 2021-06-01

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @xuzhang3 - LGTM and once the conflicts are resolved good to merge 🌵

**Comment by xuzhang3** [comment]
@katbyte  updated 😄

**Pull request merged**
